src/m3ninx/persist/reader.go

```diff
@@ -116,3 +119,58 @@ func newSafeIndexSegmentFileSetCloser(fileset IndexSegmentFileSet) io.Closer {
 	}
 	return x.NewSafeMultiCloser(closers...)
 }
+
+// NewMmapedIndexSegmentFile returns an IndexSegmentFile backed by the provided bytes and FD.
+// The returned object assumes ownership of the input fd, and mmap-ed bytes.
+func NewMmapedIndexSegmentFile(
```

Comment:
Why does this keep track of the `fd` at all? You don't need the fd to maintain an mmap

Reply:
Can you safely close the FD though?  Wasn't 100% sure of that.  Probably can I suppose?

Reply:
https://stackoverflow.com/questions/17490033/do-i-need-to-keep-a-file-open-after-calling-mmap-on-it

I suppose we can then?

Reply:
Yeah I think its fine, see our `seeker.Open()` implementation: https://github.com/m3db/m3/blob/master/src/dbnode/persist/fs/seek.go#L159

We open all the files, defer a close on all the FDs, and then mmap what we need.